qutip/core/data/kron.pyx

```diff
@@ -15,7 +15,7 @@ __all__ = [
 
 
 @cython.overflowcheck(True)
-cdef cython.numeric _mul_checked(cython.numeric a, cython.numeric b):
+cdef idxint _mul_checked(idxint a, idxint b):
```

Comment:
Why change this? Template in cython work so that that  each use of `cython.numeric` are the same type in all specialization. So it would work and be safe both way.

Reply:
Ok, cython doc says it goes up to long which can be 32 or 64 bit depending on the system so we can't rely on it.